core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java

```diff
@@ -189,7 +189,7 @@ private static BlockMetaInfo createBlockMetaInfo(
         CarbonFile carbonFile = FileFactory.getCarbonFile(carbonDataFile);
         return new BlockMetaInfo(new String[] { "localhost" }, carbonFile.getSize());
       default:
-        return fileNameToMetaInfoMapping.get(carbonDataFile);
+        return fileNameToMetaInfoMapping.get(FileFactory.getCarbonFile(carbonDataFile).getPath());
```

Comment:
please add comment why this is required? Can we add a case for Alluxio separately?

Reply:
ok